### PR TITLE
Custom render

### PR DESCRIPTION
**Pull request by vesparny**
@moeHaydar you can review

### CODE REVIEW

File: dist/koemei-search-results.js [hunk 1]
```diff
@@ -499,7 +507,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	    };
 
 	    _this.engine.search(query, sync, async);
-	    !syncCalled && sync([]);
```
Comment (moeHaydar) [inline dist/koemei-search-results.js]: @vesparny why did you remove this ?

Comment (vesparny) [inline dist/koemei-search-results.js]: @moeHaydar I don't remember but if I'm not wrong, I noticed that the check is useless because the two functions are called every time, either way.

Could you check the PR out and verify this point?

Comment (vesparny) [inline dist/koemei-search-results.js]: I meant that with or without that check, the result is always the same

File: dist/koemei-search-results.js [hunk 2]
```diff
@@ -520,7 +530,9 @@ return /******/ (function(modules) { // webpackBootstrap
 	        _this.cancel = utils.noop;
 
 	        rendered += suggestions.length;
-
+	        if (_this.options.customRendering) {
+	          return _this.options.customRendering(query, suggestions);
```
Comment (moeHaydar) [inline dist/koemei-search-results.js]: i guess we should also apply slicing here.. just as below

**Comment by moeHaydar** [comment]
@vesparny without custom rendering we have 2 functions that are called on every search
`_overwrite` and `_append`

`overwrite clears everything in the div and adds results from local memory and `_append` only appends to that div.

in the case of `customRendering` this is not the case...
if i am not mistaken, each time `customRendering` is called all the div content changes.

one solution would be to have both `customOverride` and `customAppend`..

**Comment by vohtaski** [comment]
@moeHaydar feel free to adapt the plugin,
@vesparny is on vacation for two weeks and might not be available.

ping me when you need PR-check

**Comment by vesparny** [comment]
Custom rendering just returns plain server resp. The name might be misleading

On August 10, 2015 4:00:46 PM GMT+02:00, Moe Haydar <notifications@github.com> wrote:
>@vesparny without custom rendering we have 2 functions that are called
>on every search
>`_overwrite` and `_append`
>
>`overwrite clears everything in the div and adds results from local
>memory and `_append` only appends to that div.
>
>in the case of `customRendering` this is not the case...
>if i am not mistaken, each time `customRendering` is called all the div
>content changes.
>
>one solution would be to have both `customOverride` and
>`customAppend`..
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/koemei/search-results-plugin/pull/9#issuecomment-129463856

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
